### DISCUSSION TITLE
automatic aux scanning

### DISCUSSION THREAD

**Discussion by fmassel** [retorquere/zotero-better-bibtex#3051] [Ideas]
Hello,

Currently, I use the "bbt-zotero-insert-key" function in Emacs to cite as I write; the final step of my workflow involves exporting a collection to a BibTeX file.  Between citing and exporting the collection, in Zotero,  I have to manually scan the aux file to add items to the collection that I later export.

I am looking for a way to automate this process in Zotero. Is it possible to trigger a scan of the aux file every x seconds, so that it populates the collection that will be exported to BibTeX without the need for manual intervention?

**Comment by retorquere** [comment]
[cron + curl are your friends](https://retorque.re/zotero-better-bibtex/exporting/json-rpc/index.html#collectionscanauxcollection-aux).

**Comment by fmassel** [comment]
Thanks, I think I got the idea. Probably the best way for me to go on about this is to write a function in emacs that:

1. Compiles the latex file,
2. Reads the aux and feeds it to Zotero through collection.scanAUX;
3. Exports the collection to BBT bibtex through autoexport.add (I’m note entirely sure about the translator option...).
4. Recompiles the latex + bibtex 

A bit of work an Googling, but I guess doable. 

Francesco 


–
Francesco P. Massel
Professor of Physics

University of South-Eastern Norway
Dept. of Science and Industry systems

**Reply by retorquere** [reply]
Close:

1. correct
2. you would feed it the path, not the contents
3. you would only set up auto-export once, either manually or via that API. Once set up, BBT will export for every change to the collection. An alternative is to use [pull-export](https://retorque.re/zotero-better-bibtex/exporting/pull/index.html).
